### PR TITLE
Include ByteStuffer

### PR DESCRIPTION
**Pull request by bryanoostra**
This object will be used to deal with encoding/decoding messages through usb of the v2 nohrd bike

### CODE REVIEW

File: sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.nohrd.bike.sdkv2
+
+/**
+ * This object can be used to decode/encode messages that are sent or received from the v2 Nohrd Bike through USB
+ */
+public object ByteStuffer {
```
Comment (nhaarman) [inline sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt:6]: Does this have to be public, is this actually something the user of the library needs to invoke?

Comment (bryanoostra) [reply sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt:6]: Writing/receiving custom packets of bluetooth is something the user of the library needs to do themselves. I figured doing the same for usb to match. This class just provides a way to encode/decode the data.

File: sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        return byteArrayOf(0xF1.toByte()) + stuffed + 0xF2.toByte()
+    }
+
+    public fun unstuff(input: ByteArray): ByteArray {
+        assert(input.firstOrNull() == 0xF1.toByte())
+        assert(input.lastOrNull() == 0xF2.toByte())
+
+        return input
+            .drop(1)
```
Comment (JBlaak) [inline sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt:28]: Stuffing and unstuffing should only be applied to the "body" of the packet, so the unstuffer shouldn't be the one that drops the start and end packets

File: sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                when (it) {
+                    0xF1.toByte() -> listOf(0xF3.toByte(), 0x01.toByte())
+                    0xF2.toByte() -> listOf(0xF3.toByte(), 0x02.toByte())
+                    0xF3.toByte() -> listOf(0xF3.toByte(), 0x03.toByte())
+                    else -> listOf(it)
+                }
+            }
+            .toByteArray()
+
+        return byteArrayOf(0xF1.toByte()) + stuffed + 0xF2.toByte()
```
Comment (JBlaak) [inline sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt:20]: Adding a start flag and end flag has nothing to do with stuffing. Move these flags into something that wraps your body

File: sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt (hunk 2 above)
Comment (bryanoostra) [reply sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt]: Renamed the object to UsbCodec

**Review by JBlaak — APPROVED** [review]

**Pull request merged**